lib/common/watchdog.c

```diff
@@ -245,25 +245,41 @@ pcmk_locate_sbd(void)
 long
 crm_get_sbd_timeout(void)
 {
-    const char *env_value = getenv("SBD_WATCHDOG_TIMEOUT");
-    long sbd_timeout = crm_get_msec(env_value);
+    static long sbd_timeout = -1;
 
+    if (sbd_timeout == -1) {
+        sbd_timeout = crm_get_msec(getenv("SBD_WATCHDOG_TIMEOUT"));
+    }
     return sbd_timeout;
 }
 
+long
+crm_auto_watchdog_timeout()
+{
+    long sbd_timeout = crm_get_sbd_timeout();
+
+    return (sbd_timeout <= 0)? 0 : (2 * sbd_timeout);
+}
+
 gboolean
 check_sbd_timeout(const char *value)
 {
     long st_timeout = value? crm_get_msec(value) : 0;
 
-    if (st_timeout <= 0) {
+    if (st_timeout < 0) {
+        st_timeout = crm_auto_watchdog_timeout();
+        crm_debug("Using calculated value %ld for stonith-watchdog-timeout (%s)",
+                  st_timeout, (value? value : "default"));
```

Comment:
!value should never get in here anyway ...

Reply:
true, but it's a library function, so it doesn't hurt to be safe, and it keeps the static analyzers happy

Reply:
well, sometimes static analyzers are really smart (or incredibly dumb ...) and do moan about such unreachable code as well. If !value then st_timeout=0 so it would never get in here ...

Reply:
ah, right, missed that. fixed and squashed into original commit